### PR TITLE
Window frame docs

### LINKED ISSUE #5295: sql: add support for optional frame exclusion
PR: https://github.com/cockroachdb/cockroach/pull/28262

From release notes:
> CockroachDB now supports an optional frame exclusion clause in the specification of window frames. [#28262][#28262] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #5295.
Informs #4655.

- Added window definition section
- Added window_definition diagram and parameters
- Added extended opt_frame_clause diagram and parameters
- Updated all output for Window Functions doc page
- Updated description of window functions to include OVER and WINDOW clause description

@mjibson I just saw #4655, but I agree that we need to update this page ASAP.

### CODE REVIEW

**Comment by mjibson** [comment]
Moving my review to the engineer who actually implemented these since I still don't really know how they work.

**Comment by ericharmeling** [comment]
@mjibson Thanks!!

File: v19.2/window-functions.md [hunk 1]
```diff
@@ -12,23 +12,76 @@ For a complete list of supported window functions, see [Functions and Operators]
...
+  {% include {{ page.version.version }}/sql/diagrams/window_definition.html %}
+</div>
+
+### Parameters
+
+Parameter | Description
+----------|-------------
+`window_name` | The name of the new window frame.
+`opt_existing_window_name` | An optional name of an existing window frame, defined in a different window definition.
+`opt_partition_clause`  | An optional `PARTITION BY` clause. See [Define Table Partitions](partitioning.html) for details.
```
Comment (rmloveland) [inline v19.2/window-functions.md]: I don't think the `PARTITION BY` clause we use in window functions is the same as the `PARTITION BY` syntax we introduced for geo-distributed clusters.  The window function `PARTITION BY` syntax is [also supported by PG](https://www.postgresql.org/docs/9.1/tutorial-window.html).

 Pretty sure we overloaded this phrase for the geo-distributed use case.

tl;dr: the link to "Define Table Partitions" can be removed

  @yuzefovich please shout if you disagree with the above just in case I'm misremembering somehow.

File: v19.2/window-functions.md [hunk 2]
```diff
@@ -38,112 +91,31 @@ Its operation can be described as follows (numbered steps listed here correspond
 
 1. The outer `SELECT DISTINCT(city) ... FROM rides` creates a "virtual table" on which the window functions will operate.
 2. The window function `SUM(revenue) OVER ()` operates on a window frame containing all rows of the query output.
-3. The window function `SUM(revenue) OVER (PARTITION BY city)` operates on several window frames in turn; each frame contains the `revenue` columns for a different city (Amsterdam, Boston, L.A., etc.).
+3. The window function `SUM(revenue) OVER (PARTITION BY city)` operates on several window frames in turn; each frame contains the `revenue` columns for a different city [partition](partitioning.html) (Amsterdam, Boston, L.A., etc.).
```
Comment (rmloveland) [inline v19.2/window-functions.md]: I think my comment above re: this is not that `PARTITION BY` also applies here (but we can wait for Yahor to confirm).  It's unfortunate that movr is fundamentally geo-distributed and this particular syntax is overloaded since it makes it quite confusing IMO.

File: v19.2/window-functions.md (hunk 1 above)
Comment (yuzefovich) [reply v19.2/window-functions.md]: Yes, you're right - those two are different unrelated things.

File: v19.2/window-functions.md [hunk 3]
```diff
@@ -38,112 +91,31 @@ Its operation can be described as follows (numbered steps listed here correspond
...
 
 <img src="{{ 'images/v19.2/window-functions.png' | relative_url }}" alt="Window function diagram" style="border:1px solid #eee;max-width:100%" />
 
-## Caveats
+### Caveats
 
-The most important part of working with window functions is understanding what data will be in the frame that the window function will be operating on. By default, the window frame includes all of the rows of the partition. If you order the partition, the default frame includes all rows from the first row in the partition to the current row. In other words, adding an `ORDER BY` clause when you create the window frame (e.g., `PARTITION BY x ORDER by y`) has the following effects:
+The most important part of working with window functions is understanding what data will be in the frame that the window function will be operating on. By default, the window frame includes all of the rows of the partition. If you order the partition, the default frame includes all rows from the first row in the partition of the current row. In other words, adding an `ORDER BY` clause when you create the window frame (e.g., `PARTITION BY x ORDER by y`) has the following effects:
```
Comment (rmloveland) [inline v19.2/window-functions.md]: I think the previous use of "to" was correct?  As in "the window frame extends from row A to row B"?  Unless I'm fundamentally misunderstanding something, which is totes possible!

File: v19.2/window-functions.md [hunk 4]
```diff
@@ -38,112 +91,31 @@ Its operation can be described as follows (numbered steps listed here correspond
...
 
 - It makes the rows inside the window frame ordered.
 - It changes what rows the function is called on - no longer all of the rows in the window frame, but a subset between the "first" row and the current row.
 
-Another way of saying this is that you can run a window function on either:
+Another way of saying this is that you can run a window function on:
```
Comment (rmloveland) [inline v19.2/window-functions.md]: I think this is still an either/or case unless I was wrong the first time around and something has changed?  If so please let me know.  But if not, removing "either" makes it seem like this is AND instead of XOR

File: v19.2/window-functions.md (hunk 1 above)
Comment (ericharmeling) [reply v19.2/window-functions.md]: Ahh, OK. That makes more logical sense. Thanks Rich!

File: v19.2/window-functions.md [hunk 5]
```diff
@@ -38,112 +91,31 @@ Its operation can be described as follows (numbered steps listed here correspond
...
-|       | vehicle_id) REFERENCES vehicles (city, id),                              |
-|       |     INDEX rides_auto_index_fk_vehicle_city_ref_vehicles (vehicle_city    |
-|       | ASC, vehicle_id ASC),                                                    |
-|       |     FAMILY "primary" (id, city, vehicle_city, rider_id, vehicle_id,      |
-|       | start_address, end_address, start_time, end_time, revenue),              |
-|       |     CONSTRAINT check_vehicle_city_city CHECK (vehicle_city = city)       |
-|       | )                                                                        |
-+-------+--------------------------------------------------------------------------+
-~~~
+{{site.data.alerts.callout_info}}
```
Comment (rmloveland) [inline v19.2/window-functions.md]: This could be a green "tip" box (they're rare!)

File: v19.2/window-functions.md [hunk 6]
```diff
@@ -263,30 +229,30 @@ To see which customers have taken the most rides while generating the most reven
...
+  Judy White       |              15 |              818.00
+  Tony Ortiz       |              15 |              724.00
+  Cindy Medina     |              14 |              740.00
+  Tyler Dalton     |              13 |              786.00
+  Samantha Coffey  |              13 |              758.00
+  Nicole Mcmahon   |              13 |              696.00
 (10 rows)
 ~~~
 
+Note that in the query above, a `WINDOW` clause defines the window frame, and the `OVER` clauses reuse the window frame.
```
Comment (rmloveland) [inline v19.2/window-functions.md]: Noice
